### PR TITLE
[JENKINS-61136] Allows URL as input for readJSON

### PR DESCRIPTION
**Pull request by eXpire163**
readJSON also allows now to read from an URL

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStepExecution.java [hunk 1]
```diff
@@ -85,6 +87,9 @@ protected Object doRun() throws Exception {
         if (!isBlank(step.getText())) {
             json = JSONSerializer.toJSON(step.getText().trim());
         }
+		if (!isBlank(step.getUrl())){
```
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStepExecution.java]: Extra indentation

File: src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStepExecution.java [hunk 2]
```diff
@@ -133,5 +138,35 @@ private boolean isNull(Object value) {
         }
         return false;
     }
+	
+	private String readStringFromURL(String requestURL) throws IOException {
+        URLConnection      connection = new URL(requestURL).openConnection();
+        connection.setRequestProperty("Content-Type", "application/json; utf-8");
+        connection.setRequestProperty("User-Agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:55.0) Gecko/20100101 Firefox/55.0");
```
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStepExecution.java]: This looks hacky

File: src/main/resources/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStep/help-url.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<!--
+  ~ The MIT License (MIT)
+  ~
+  ~ Copyright (c) 2016 Nikolas Falco, Christian Wiedemann
```
Comment (rsandell) [inline src/main/resources/org/jenkinsci/plugins/pipeline/utility/steps/json/ReadJSONStep/help-url.html]: ??

**Review by jglick — CHANGES_REQUESTED** [review]
Would be a serious security vulnerability. Do not merge.

**Pull request closed without merging**